proposals/2024-01-26_classic-histograms-stored-as-native-histograms.md

```diff
@@ -96,9 +97,13 @@ Scenarios provided that the native histograms feature is enabled:
 * Resulting samples at scrape time reuse the existing data structures for native histograms, but use a special schema number to indicate custom buckets. A sample would have either custom bucket definitions or exponential schema and buckets, but not both.
 * If the histogram has exponential buckets during scrape then only the exponential buckets are kept and the custom buckets would be dropped.
 * If all bucket counters and the overall counter of the histogram is determined to be a whole number, use integer histogram, otherwise use float histogram.
-  * In the Prometheus text exposition format values are represented as floats. The dot (`.`) is not mandatory.
+  * In the Prometheus text exposition format values are represented as floats. The dot (`.`) is not mandatory. After parsing the value `x` if the GO statement `x == math.Trunc(x)` is true, then use integer counters, otherwise switch to floats.
   * In the OpenMetrics text exposition [format](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#numbers) floats and integers are explicitly distinguished by the dot (`.`) sign being present or not in the value.
   * In the Prometheus/OpenMetrics ProtoBuf [format](https://github.com/prometheus/client_model/blob/d56cd794bca9543da8cc93e95432cd64d5f99635/io/prometheus/client/metrics.proto#L115-L122) float and integer numbers are explicitly transferred in different fields.
+* Conversion of classic buckets defined by `[le1, le2, ..., leM, leN]` upper boundaries:
```

Comment:
Actually the "conversion" should go to Reading via PromQL section since in storage we don't really care how the boundaries are interpreted, just that there are some buckets with counts. So the "conversion" at scrape is to store the boundaries in a new data field and the counts in positive buckets.